packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,78 @@
+import { difference, uniq } from 'lodash'
+
+export default class {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async mergeInto({ sources, target, force }) {
+    const { _xo } = this
+    const targetHost = _xo.getObject(target.master)
+    const _sources = []
+    const sourcePatches = {}
+    let source
+    let sourceId
+    let sourceHost
```

Comment:
Declare them in the `for`.

Reply:
No, it's exactly the same performance wise.

But I find it cleaner to declare the variables only when necessary, with the minimal scope possible, which helps avoid some classes of bugs.

Also, it will allow you use `const` instead of `let` in some places.